code/handlers/ldap.q

```diff
@@ -45,34 +69,38 @@ login:{[user;pass]                                              / validate login
       update attempts:0, blocked:0b from `.ldap.cache where user=user];
   ];
 
+  /authenticate:{[dict].ldap.bind[.ldap.sessionID;(enlist `$dict[`bind_dn])!enlist `$dict[`pass]]}
+
   authorised:$[all (                                            / check if previously used details match
     incache[`success];                                          / previous attempt was a success
     incache[`time]>.z.p-.ldap.checktime;                        / previous attemp occured within checktime period
     incache[`pass]~np:md5 pass                                  / same password was used
   );
     1b;
-    @[{.ldap.authenticate[x]`success};dict;0b]                  / attempt authentication
+    authenticate:{[dict].ldap.bind[.ldap.sessionID;(`dn;`cred)!(dict[`bind_dn];`$dict[`pass])]} [dict]                 / attempt authentication
   ];
  
-  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; (1+0^incache`attempts;0) authorised;authorised;0b);  / upsert details of current attempt
+  
 
-  $[authorised;                                                 / display authentication status message
+  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; (1+0^incache`attempts;0) authorised[`ReturnCode] ;$[authorised[`ReturnCode]~0i;1b;0b];0b);  / upsert details of current attempt
```

Comment:
I think indexing into `authorised` like this will fail if the user was in the cache? in that case, `authorised` will currently be a true boolean (`1b`), so we can't index like this (easiest solution might be to change the `1b` on line 53 to a dictionary like ``enlist[`ReturnCode]!enlist 0i``)

Also, ``$[authorised[`ReturnCode]~0i;1b;0b]`` - the conditional is redundant, you can just use ``authorised[`ReturnCode]~0i`` which will return a boolean anyway